src/dex-ui/hooks/useSwapData.tsx

```diff
@@ -17,12 +17,14 @@ export const useSwapData = (refreshInterval = 0) => {
     swap,
   }));
   const walletAccountId = wallet.walletData.pairedAccounts[0];
-  const { transactionState, fetchFee, fetchSpotPrices, getPrecision } = swap;
+  const { transactionState, fetchFee, fetchSpotPrices, getPrecision, fetchTokenPairs } = swap;
 
   const fetchSwapDataOnLoad = useCallback(async () => {
     getPrecision();
+    // Get Token pairs (called from store)
     await Promise.allSettled([fetchFee(), fetchSpotPrices()]);
```

Comment:
You could add `fetchTokenPairs()` to the array passed into the first `Promise.allSettled` to asynchronously fetch the fee, spotPrice, and token pairs. 

`await Promise.allSettled([fetchFee(), fetchSpotPrices(), fetchTokenPairs()])`

That may help reduce the time it takes to fetch all of the swap data.

Reply:
Done